src/middlewared/middlewared/service/core_service.py

```diff
@@ -207,6 +207,26 @@ def get_jobs(self, app, filters, options):
         ], filters, options)
         return jobs
 
+    @no_authz_required
+    @accepts(Int('id'), Str('filename'), Bool('buffered', default=False))
+    @pass_app(rest=True)
+    async def job_download_logs(self, app, id_, filename, buffered):
+        """
+        Download logs of the job `id`.
+
+        Please see `core.download` method documentation for explanation on `filename` and `buffered` arguments,
+        and return value.
+        """
+        if app is None:
+            job = self.middleware.jobs[id_]
+        else:
+            job = self.__job_by_credential_and_id(app.authenticated_credentials, id_)
```

Comment:
If I'm understanding correctly, if a bogus id is given then `job` object will be `NoneType`?

Reply:
No, `__job_by_credential_and_id` returns a job or raises an exception

Reply:
If a bogus ID is given then a KeyError will be raised in that line

Reply:
okay but `__job_by_credential_and_id` raises `CallError`. Since this is for the UI team, should we check with them to see if they want this endpoint to wrap these potential errors in `ValidationErrors`?